src/App.tsx

```diff
@@ -1,14 +1,71 @@
+/* eslint-disable react-hooks/rules-of-hooks */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './TodoList';
+import { Footer } from './Footer';
+import { Errors } from './types/Errors';
+import { FilterBy } from './types/FiilterBy';
+
+type FilterTheTodos = (todos: Todo[], filterBy: FilterBy) => Todo[];
+
+const getFilteredTodos: FilterTheTodos = (todos, filterBy) => {
+  let filteredTodos = todos;
+
+  if (filterBy !== FilterBy.All) {
+    filteredTodos = filteredTodos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return todo.completed === false;
+        case FilterBy.Completed:
+          return todo.completed === true;
+        default:
+          throw new Error('Unknown filter type');
+      }
+    });
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        setErrorMessage(Errors.Load);
+        throw error;
+      });
```

Comment:
```suggestion
      .catch(error => {
        setErrorMessage(Errors.Load);
      });
```
Don't throw errors if you don't handle them at the next step